library/src/main/java/com/datatorrent/lib/appdata/schemas/SnapshotSchema.java

```diff
@@ -246,6 +247,26 @@ private void initialize() throws JSONException
     schemaJSON = this.schema.toString();
   }
 
+  public void setTags(Set<String> tags)
+  {
+    if (tags == null || tags.isEmpty())
+      return;
+
+    JSONArray tagArray = new JSONArray();
+
+    try {
+      int index = 0;
+      for (String tag : tags)
+        tagArray.put(index++, tag);
+
+      schema.put(FIELD_SCHEMA_TAGS, tagArray);
+    } catch (JSONException e) {
+      Preconditions.checkState(false, e.getMessage());
```

Comment:
There are several constructors, add this parameter to the constructors will add more constructors( one for no tags, one for tags ). So I think maybe use setTags more clear.